File: app/src/main/java/in/hridayan/ashell/UI/Transitions.java
package in.hridayan.ashell.UI;

import android.view.View;
import com.google.android.material.transition.MaterialContainerTransform;
import android.graphics.Color;
import android.view.ViewGroup;
import androidx.transition.TransitionManager;

public class Transitions {

  public static void materialContainerTransformViewToView(View startView, View endView) {
    MaterialContainerTransform transform = new MaterialContainerTransform();

    transform.setStartView(startView);
    transform.setEndView(endView);
    transform.addTarget(endView);

    // Optionally add a curved path to the transform
    // transform.setPathMotion(new MaterialArcMotion());

    transform.setScrimColor(Color.TRANSPARENT);

    TransitionManager.beginDelayedTransition((ViewGroup) startView.getRootView(), transform);

    startView.setVisibility(View.GONE);
    endView.setVisibility(View.VISIBLE);
  }
}

File: app/src/main/java/in/hridayan/ashell/ui/Transitions.kt
package `in`.hridayan.ashell.ui

import android.graphics.Color
import android.view.View
import androidx.transition.TransitionManager
import com.google.android.material.transition.MaterialContainerTransform

object Transitions {
    @JvmStatic
    fun materialContainerTransformViewToView(startView: View, endView: View) {
        val transform = MaterialContainerTransform()

        transform.startView = startView
        transform.endView = endView
        transform.addTarget(endView)

        // Optionally add a curved path to the transform
        // transform.setPathMotion(new MaterialArcMotion());
        transform.scrimColor = Color.TRANSPARENT

        TransitionManager.beginDelayedTransition(
            (startView.getRootView() as android.view.ViewGroup),
            transform
        )

        startView.visibility = View.GONE
        endView.visibility = View.VISIBLE
    }
}
